src/matrix/core/endutils.nim

```diff
@@ -39,7 +39,7 @@ func addQuery*(e: var Endpoint, query: openArray[(string, string)]): void =
 func build*(
   endpoint: EndpointDraft,
   homeserver: Uri,
-  params: varargs[(string, string)] = []): Endpoint =
+  urlParams, queryParams: varargs[(string, string)] = []): Endpoint =
```

Comment:
Do the other functions work normally with this change?

Reply:
kinda, you need to specify whether they are `urlParams` or `queryParams`, but I added a new test for that

Reply:
Okay, I will look at the functions as well to make sure they're all good.